File: util/src/main/java/kr/poturns/util/media/BaseDirInfo.java
package kr.poturns.util.media;


import org.json.JSONObject;

public abstract class BaseDirInfo<T extends BaseInfo> extends Info {
    public T firstInfo;

    @Override
    public JSONObject toJSON() {
        try {
            return new JSONObject()
                    .put("dirName", dirName)
                    .put("firstInfo", firstInfo.toJSON());

        } catch (Exception e) {
            return new JSONObject();
        }

    }

}

File: UnityGUI/Assets/BridgeApi/Media/BaseDirInfo.cs
using System.Collections.Generic;

namespace BridgeApi.Media
{
    public interface BaseDirInfo<T> where T : BaseInfo
    {
        /// <summary>
        /// 디렉토리 이름
        /// </summary>
        string DirName { get; set; }

        /// <summary>
        /// 디렉토리에 존재하는 Info중 첫번째 Info
        /// </summary>
        T FirstInfo { get; set; }

        /// <summary>
        /// 디렉토리에 존재하는 모든 Info의 리스트를 반환한다.
        /// </summary>
        /// <returns>디렉토리에 존재하는 모든 Info의 리스트</returns>
        List<T> GetInfoListInDirectory();

    }
}

